0010-endpoint-metadata-extension/README.md

```diff
@@ -0,0 +1,658 @@
+    }
+  },
+  "additionalProperties": false
+}
+```
+Examples:
+- get metadata without message identifier
+
+```json
+{}
+```
+- get metadata with message identifier
+
+```json
+{
+  "id": 42
+}
+```
+
+#### Get metadata keys response
+
+The server MUST respond to the metadata get request by publishing the response message according to the [request/response design pattern defined in 1/KP](/0001-kaa-protocol/README.md#requestresponse-pattern).
+The extension-specific resource path part format is:
+
+  `<endpoint_token>/get/keys/status`
+
+The response payload MUST be a JSON-encoded object with the following [JSON schema](http://json-schema.org/) ([file](./schemas/get-metadata-keys-response.schema.json)):
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "10/EPMDX get metadata keys response",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "integer",
+      "description": "The message identifier used to match server response to the request"
+    },
+    "keys": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "minLength": 1,
+        "description": "The endpoint metadata key"
+      },
+      "uniqueItems": true,
+      "description": "The set of endpoint metadata keys"
+    },
+    "statusCode": {
+      "type": "integer",
+      "minimum": 1,
+      "description": "Status code based on HTTP status codes"
+    },
+    "reasonPhrase": {
+      "type": "string",
+      "minLength": 1,
+      "description": "A human-readable string explaining the cause of an error (if any). `OK` if request was successful"
+    }
+  },
+  "required": [
+    "statusCode",
+    "reasonPhrase"
+  ],
+  "additionalProperties": false
+}
+
+```
+
+  where
+  - `id` MUST match that in the request
+
+Examples:
+
+- Endpoint metadata successfully retrieved
+
+```json
+{
+  "id":42,
+  "statusCode":200,
+  "reasonPhrase":"OK",
+  "keys":[
+    "name",
+    "location",
+    "deviceModel"
+  ]
+}
+```
+- No endpoint metadata found
+
+```json
+{
+  "id":42,
+  "statusCode":404,
+  "reasonPhrase":"Endpoint metadata not found"
+}
+```
+
+### Delete metadata key
+#### Delete metadata key request
+
+Client MUST send requests with the following extension-specific [resource path](/0001-kaa-protocol/README.md#resource-path-format) part in order to receive the metadata key-value pairs:
+
+  `<endpoint_token>/delete/key`
+
+  where `<endpoint_token>` identifies the endpoint.
+
+The request payload MUST be a JSON-encoded object with the following [JSON schema](http://json-schema.org/) ([file](./schemas/delete-metadata-key-request.schema.json)):
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "10/EPMDX delete metadata key request",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "integer",
+      "description": "The message identifier used to match server response to the request"
+    },
+    "keyName": {
```

Comment:
Currently we support only deletion of one key, so `keys` is not suitable name. We can rename this field to `key`